lib/src/session.rs

```diff
@@ -590,6 +597,38 @@ impl Session {
         &self.dictionaries
     }
 
+    // ----- GeoIP API -----
+
+    pub fn geoip_lookup(&self, addr: &IpAddr) -> String {
+        match self.geoip_mapping.get(addr) {
+            Some(geoip) => geoip.to_string(),
+            None => String::from(
```

Comment:
Can we have this instead be a default value in the mapping for 127.0.0.1 and the ipv6 equivalent that can be overwritten in the config. By that I mean when the mapping is first initialized in `GeoIPMapping::new()` this entry is included for localhost and can be overwritten after when the `fastly.toml` is read.

The reasoning being that if a customer is trying to test doing an IP lookup for a value that isn't available they will always get a value now which can be confusing since there are cases where on C@E it will return a `None` value.

Reply:
I'm just thinking if we could somehow even disable the placeholder for loopback addresses as well? Perhaps we can make it so that it returns the placeholder address only in the case when you aren't reading addresses from JSON or reading them from TOML, effectively making it work only when you use the absolute default geolocation settings. Or perhaps we can have this behind a settings flag in the TOML?

For example, sometimes you may want to test what happens to your code when the IP address of the incoming request yields no result.

Reply:
@bbutkovic am I understanding correctly that you'd want to add an option to remove the default loopback value?

Reply:
Yes, precisely, if you choose to provide your own resolving through a JSON or TOML you must specify the result for loopback as well, otherwise it will be a "failed" lookup.

Essentially, the default would only work if you don't use TOML or JSON.

Reply:
Ah okay! I think what we want is sort of a hybrid approach

- By default we provide a loopback address default
- If it's overwritten by a value in the toml/json use that
- You can disable the default by setting a boolean field in the config under `local_server.geolocation_mapping.use_default_loopback` to false and this defaults to true

I think this will let us document the behavior better and allow people to opt into turning it off or overriding it! Does that sound good to you?

Reply:
That sounds good!

Reply:
Great! I think this is probably the last thing we'll need. I'll review it whenever you get it in (just give me an @ ping)